addons/assigngear/functions/fnc_assignGear.sqf

```diff
@@ -99,59 +99,77 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 14: { // scope
+            case IDX_PRIMARY_MAGAZINE: { // primarymagazine
+                {
+                    _unit addPrimaryWeaponItem _x;
+                } forEach _x;
+            };
+            case IDX_PRIMARY_SCOPE: { // scope
                 private _scope = selectRandom _x;
                 if !(_scope isEqualTo '') then {_unit addPrimaryWeaponItem _scope};
             };
-            case 15: { // bipod
+            case IDX_PRIMARY_BIPOD: { // bipod
                 private _bipod = selectRandom _x;
                 if !(_bipod isEqualTo '') then {_unit addPrimaryWeaponItem _bipod};
             };
-            case 16: { // attachment
+            case IDX_PRIMARY_ATTACHMENT: { // attachment
                 private _attachment = selectRandom _x;
                 if !(_attachment isEqualTo '') then {_unit addPrimaryWeaponItem _attachment};
             };
-            case 17: { // silencer
+            case IDX_PRIMARY_SILENCER: { // silencer
                 private _silencer = selectRandom _x;
                 if !(_silencer isEqualTo '') then {_unit addPrimaryWeaponItem _silencer};
             };
-            case 18: { // secondaryWeapon
+            case IDX_SECONDARY_WEAPON: { // secondaryWeapon
                 private _weapon = selectRandom _x;
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 19: { // secondaryAttachments
+            case IDX_SECONDARY_MAGAZINE: { // secondarymagazine
+                {
+                    _unit addSecondaryWeaponItem _x;
+                } forEach _x;
+            };
+            case IDX_SECONDARY_ATTACHMENT: { // secondaryAttachments
                 {_unit addSecondaryWeaponItem _x} forEach _x;
             };
-            case 20: { // sidearmweapon
+            case IDX_SIDEARM_WEAPON: { // sidearmweapon
                 private _weapon = selectRandom _x;
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 21: { // sidearmattachments
+            case IDX_SIDEARM_MAGAZINE: { // sidearmmagazine
+                {
+                    _unit addHandgunItem _x;
+                } forEach _x;
+            };
+            case IDX_SIDEARM_ATTACHMENT: { // sidearmattachments
                 {_unit addHandgunItem _x} forEach _x;
             };
-            case 22: { // Unit traits
+            case IDX_TRAITS: { // Unit traits
                 {
                     [_unit, _x] call FUNC(setUnitTrait);
                 } forEach _x;
             };
-            case 23: { // code
+            case IDX_CODE: { // code
                 _unit call compile _x;
             };
         };
     };
 } forEach _loadoutArray;
 
+// Force reload the unit to pull mags into all weapons if not specified in each of the weapons.
+reload _unit;
```

Comment:
Forces a reload to maintain backwards compat of units spawning with all weapons loaded.

Reply:
I appreciate there is a bit of novelty to this approach but it does result in a reload animation and only the currently held weapon is loaded, so if a player has a primary weapon and a side-arm: the side arm will not be loaded. Also in the editor the magazine will not be displayed.

I think a better solution here is to simply change the order in which a weapon is given. e.g.
- if `xMagazine` is set, then add the weapon before the general magazines are added. Then run the line of code to add the magazine. Then subsequently add the general magazines. 
- if `xMagazine` is not set, add the weapon after the general magazines are added (so they have the opportunity to auto-load a mag from the general magazines.

Reply:
[I was lied to](https://community.bistudio.com/wiki/reload#:~:text=Description%3A-,Reload%20all%20weapons.,-Groups%3A) 😂

Reply:
I guess given the amount of functionality dependent on ordering in that list we can add a little more, the alternative would be to pull all the required ordering out of those indexes and make it explicit in code.

Reply:
Is this what you were thinking @Sniperhid?

Reply:
Yeah this is much better, but slight issue. Currently only primary weapon will be assigned. Because the index of magazine comes before the secondary weapon, as the indexes are being looped through `_secondaryWeapon`, and `_sidearmWeapon` remain empty when the magazines are processed. A fix for this is to move secondary/sidearm (and their magazines) index before the magazines.

Reply:
They are before the `_magazines` https://github.com/TMF3/TMF/pull/420/files#diff-5c206fa83b8b2ee52e21e3c6f57bbdde8836e5ce9d609c0bf3f1ebb73cd416e6R58-R71

Is this a bug you're noticing?

Reply:
Works fine on my machine 😅

Reply:
> As this just loops through the array and executes by index order. Which is necessarily the order the case statements are written in.

My understanding is that the code iterates through each element in the array in element order. Then based on the current index of the element, executes the relevant case statement based on the order. 

The element order of the array is determined by the indexes laid out in `script_component.hpp` not the order of the case statements, which is part of the refactor to centralise those constants. So in theory, it should execute the magazine and weapon case statements before the magazines according to that code. Which I believe I'm seeing in my testing.

Reply:
We're on the same page, turns out I was reading an old version of the indexes. Bug is elsewhere on legacy loadouts.

Reply:
> Bug is elsewhere on legacy loadouts.

Oh, I thought it was on new loadouts, let me dig further.